main/asset_loaders/fs_wsg.c

```diff
@@ -108,18 +108,36 @@ bool loadWsgInplace(cnfsFileIdx_t fIdx, wsg_t* wsg, bool spiRam, uint8_t* decomp
     }
 
     // Save the decompressed info to the wsg. The first four bytes are dimension
-    wsg->w = (decompressedBuf[0] << 8) | decompressedBuf[1];
-    wsg->h = (decompressedBuf[2] << 8) | decompressedBuf[3];
+    uint16_t newW = (decompressedBuf[0] << 8) | decompressedBuf[1];
+    uint16_t newH = (decompressedBuf[2] << 8) | decompressedBuf[3];
+
+    // If there is an existing buffer and it doesn't match, free it
+    if (wsg->px && (wsg->h != newH || wsg->w != newW))
```

Comment:
If we wanted to be really cute, we dont need to check W and H separately. What we really need to check is if w*h==w*h, i.e. the total size. I think the cases where that math checks, but the dimensions change are fleetingly rare